### ISSUE TITLE
[FR] Add ability to record on any movement

### ISSUE THREAD

**Issue by ljmerza** [blakeblackshear/frigate#803]
My Frigate instance has way too many false positives so I've had to raise the threshold but now I get a lot of false negatives. Since frigate only records either "clips" on object detection or 24/7, there's no way to realistically sift through videos that are actually useful. I either miss important events on clips from false negatives or have to look through 24/7 recordings.

I think a good balance would be to add the ability to record on any movement.

**Comment by blakeblackshear** [comment]
If you were struggling to sift through the false positives with detection, this won't help. Recording on all motion would just increase the number of false positives.

**Comment by ljmerza** [comment]
I struggle with false negatives not false positives... That's the point of recording on motion. It's 24/7 recording, how could I know where to look if someone stole something off my back porch, I didn't notice for 3 days, and the object detection didn't pick it up? You are talking about 3 days worth of 24/7 recordings I have to sift through now to find where it was stolen. I want to be able to record on all motion, then false negatives wont be as important to miss. Recording on motion is an absolute standard on NVR systems anyways.

**Comment by muzzak123** [comment]
I'm not sure if this is the same thing , but I would like to be able to capture a pic and clip based on ANY motion in a defined zone.
It seems like Frigate will only do this it it recognises an object. For example I would like to be able to recognise if my garage camera "sees" the garage door open. I've check the camera in Frigate with Motion Boxes turned on and it does detect the motion but obviously doesnt recognise it as an object.  Perhaps an object of "Motion" could be used ? 
I know FFmpeg Motion https://www.home-assistant.io/integrations/ffmpeg_motion/ will do this, but it would be nice to have it all bundled into the amazing Frigate UI.

**Comment by ronschaeffer** [comment]
+ 1 for this. It would be great for consolidating camera management with Frigate awesome UI/HA integration.

**Comment by chrisgott** [comment]
i do have similar problems at nighttime (no ir able cam). i guess it will detect the motion, but cant detect either person or car, so there is no alerting. advising burglars to visit me at daytime only is maybe not the best option, so im asking if it is possible somehow to set a alert via mqtt on motion only? or maybe set score lower so that humanoid shadows will trigger?

**Comment by blakeblackshear** [comment]
You can use the detection fps sensor to alert on motion if you want.

**Comment by ljmerza** [comment]
Alert on movement might be too much. Can I record on movement instead?

**Comment by blakeblackshear** [comment]
If you could, this issue would already be closed.

**Comment by chrisgott** [comment]
> You can use the detection fps sensor to alert on motion if you want.

it increases from 0 to 0.2 fps on first motion (person walking) and stays long at this value befor it falls back to 0. a second following motion (moving car) following up just seconds after leads to no change. anyways i'll give that a try cause its better than nothing.

**Comment by ljmerza** [comment]
Don5 close it. Still an issue

**Comment by osos** [comment]
I was looking for this as well.

Would love to just track "motion" in particular zones, and objects in other zones.

**Comment by ljmerza** [comment]
Don't close

**Comment by blakeblackshear** [comment]
0.9.0 will lay the groundwork for this. In a future release, motion will be available as an option for the recordings retention policy.

**Comment by muzzak123** [comment]
Great news

**Comment by Numline1** [comment]
If anyone's interested, there's a bit clunky, but pretty fair workaround - motionEye addon can already record on any movement and it also supports re-streaming. You could temporarily-ish use that for motion detection and then feed the data to Frigate for object detection. Not the best solution ever, but it should work fairly well.

**Comment by muzzak123** [comment]
@Numline1 I'm not sure what you mean. Perhaps you can elaborate. From my perspective Frigate can detect the motion by changes in the fps sensor, but it doesnt record it because it doesnt recognise the object. I want to be able to record even if the object is not recognised by Frigate

**Comment by Numline1** [comment]
@muzzak123 I'm sorry, I kinda assumed everyone's using Frigate as an Home Assistant addon, which, obviously, is not true. Either way - motionEye is a separate software (motionEye is the frontend, motion is the backend - https://motion-project.github.io). Also, keep in mind, this is a workaround / possible solution for a problem, not a Frigate integration or anything like that.

As I've mentioned in my previous comment, you can temporarily (or permanently) install `motion` to deal with your motion detection/recording and then re-stream the RTMS into Frigate to deal with object detection. Once Frigate has its own motion detection in 0.9.x, you can get rid of `motion` (or keep it, if you'd like).

I'd also like to note that I'm in no way associated with either motion or Frigate projects, hopefully the Frigate author doesn't mind me mentioning a 3rd party project. It's just something that works well for me, since I find both projects to be pretty awesome.

**Comment by cianpdx** [comment]
could a workaround be to set threshold for an "object" to be so low that it always gets a positive from the detection? then it is saved. 

but you can run your automations off the person or automobile detection, which has normal thresholds

**Comment by AdmiralNemo** [comment]
That's similar to my workaround.  There is always a car in my camera's field of view,so I have Frigate set to record when it detects a car.  It works since Frigate only cares that there was motion and that a car was detected, not necessarily that the car was in motion.

**Comment by Junkbite** [comment]
> could a workaround be to set threshold for an object you don't care about (hairdryer) to be so low that it always gets a positive from the detection? then it is saved. 
> 
> but you can run your automations off the person or automobile detection, which has normal thresholds
> 
> 

No because then it would think all types of stationary items are objects and record constantly. If you set your threshold really low then all types of stuff in the picture will trigger. A garden hose sitting in my back yard would trigger as a person when I set the threshold low even though it wasn't moving. This would not work as a substitute.

**Comment by Ataman** [comment]
New frigate user here. My installation has difficulty detecting cars (or people) in the darker areas at night. Problem is, the suspicious activity we're trying to record mostly happens at night.

Having the ability to mark a zone that records motion even without detection but only at specifics times would be a great fallback. Yes, it would cause false positives but since it's night it should be manageable.

**Comment by blakeblackshear** [comment]
@Ataman more often than not, the issue here is that motion doesn't register at night because the contrast is too low. There probably isn't any motion to save recordings for. I would suggest giving the new beta a try to see if that performs better at night.

**Comment by Ataman** [comment]
@blakeblackshear According to the debug view it can clearly see that there is _something_, just not clear enough to reach any meaningful detection threshold. I can't lower the threshold however because that would cause too many false positives during the day.

My guess was that the edge model isn't reliable enough for our setup at night but I'll try the beta next time I have access to the system.

Thank you for your time.

**Comment by fcasarsa** [comment]
I have outdoor flood lights and porch lights that i would like to turn on when a generic motion ( eg car, cat, monster, alien.. ) is detected.

Is there a way to do it in the most recent release? 

Thanks.

**Comment by doug62** [comment]
Plus 1 this request, or provide specific config value to change (I find that the documentation is not complete but love the concept and thanks)....  I use this for mouse detecting/census in my house - anything that moves is a critter...

**Comment by mateuszdraco** [comment]
+1 on this one :)

**Comment by blakeblackshear** [comment]
The latest 0.10.0 Beta 6 supports the ability to retain only recording segments with motion.

**Comment by doug62** [comment]
@blakeblackshear May I ask where motion is enabled/configured in the app? Thought i reviewed recent PRs/Docs and couldn't find it - thanks.

**Comment by blakeblackshear** [comment]
Preview for new docs is here: https://deploy-preview-1927--frigate-docs.netlify.app/configuration/index

**Comment by BBaoVanC** [comment]
> The latest 0.10.0 Beta 6 supports the ability to retain only recording segments with motion.

I used this setting, but it doesn't seem to create events for motion, which is what I was hoping to do. Is there a way to do that? I have it recording 24/7 and I can't just sift through all of the recordings to try and find the motion event.

**Comment by blakeblackshear** [comment]
It doesn't generate events. Maybe you just have tons of motion, but when I go back and watch recordings, most hours only have a few minutes of footage that was retained. I can review a night's worth of footage in a minute or two now.

**Comment by BBaoVanC** [comment]
I have it recording 24/7 like this:

```yaml
record:
  enabled: True
  retain:
    days: 2
    mode: all
  events:
    retain:
      default: 30
      mode: motion
```

I'd like it to record 24/7 but still want to be able to know when there's motion, even if there's not a detected object. Yesterday there was a cardinal but it wasn't able to be picked up by object detection and sifting through the 24/7 recording will be a pain.

**Comment by Minglarn** [comment]
I am kinda looking for the same idea... 
It would be nice to sort the event menu by "OBJECTS" or "MOTION" ..
See this: https://github.com/blakeblackshear/frigate/issues/2448

**Comment by LucaDiba** [comment]
Hi, I have a question regarding the CPU usage when Frigate is looking for any motion.
I'm currently not able to find a Coral anywhere and I cannot use all the fun Frigate features (I can't wait to be able to buy a Coral and start using Frigate as I should). I'm running Frigate 0.10.0 Beta 6 on a Raspberry Pi 4 B 8GB (it's dockerized and I'm running other things on the same RPi).

I'm recording a low quality stream 24/7, but I record in high quality when motion is detected. I noticed that there is no problem at all while recording 24/7 only, however sometimes the video lags when there is something going on (e.g., a car or a person passes). I think that's because of the object detecion, which is done on the CPU and blocks everything until the object is detected. The fact that Frigate records only when nothing happens and sometimes stops recording when something happens is not the best as you can imagine 😅

So my question is: is it possible to perform motion detection with the object detection disabled?
I'm using the following configuation:
   ```
record:
     enabled: True
     retain:
       days: [number of days]
       mode: motion
```

The problem is that the video is saved when motion occurs only if `detect` is enabled: if I set `detect: enabled: False` then Frigate does not save the parts when motion is detected.

Thanks!

**Comment by blakeblackshear** [comment]
I don't think motion detection alone works currently, but it could. I will look into adding it into the beta.

**Comment by doug62** [comment]
I'm not sure motion works either, I haven't completely tried to get it working, I think I have it configured but I don't see it on the ui,  regarding performance - attached is a utilization chart of my stack, 4 cams, 2-5 fps, seems it wants more than for cores or 1 core/cam,  I noticed it was much happier when I gave it 15 cores, right now my 4 cores is affecting accuracy. 
![image](https://user-images.githubusercontent.com/5632034/150653855-fb14b97c-622c-4deb-9acb-dbc52fd1f196.png)

**Comment by blakeblackshear** [comment]
@LucaDiba With 0.10.0 Beta 9, it is now possible to record motion with detect disabled.

**Comment by doug62** [comment]
@LucaDiba  Thanks, I believe I have motion enabled, just updated to Beta 9 - can't see how/where motion recordings are happening?

**Comment by blakeblackshear** [comment]
Motion does not generate events in the events tab, but it will limit recordings to segments with motion. You can go back and review recordings and it will only show you sections of video with motion. The following config will keep 7 days of recordings, but only segments with actual motion:

```yaml
retain:
  enabled: True
  retain:
    days: 7
    mode: motion # <--- default
```

This works even when detect is disabled.

**Comment by LucaDiba** [comment]
@blakeblackshear thanks a lot!!! It'll help keeping the resources consumption low if not needed.

I've updated to beta 9 but unfortunately I still have the same problem as before.
At this point I think that the problem is just that I'm not using a Coral - do you want me to open a new issue to investigate anyway?

**Comment by blakeblackshear** [comment]
> do you want me to open a new issue to investigate anyway?

Sure. I may be able to help you make some adjustments.

**Comment by hawkeye217** [comment]
Adding on to this thread... I just upgraded to 0.10.0 and I love the option to be able to record on motion.

Is there a way now in 0.10.0 to disable everything for a camera, including motion? I have an automation that disables an indoor camera's detection when someone is home, but now that camera shows up in the Birdseye view whenever there's motion on it. It did not do that in 0.9.x.  

The camera is a PTZ that turns toward the wall when someone is home, and I'm seeing an occasional motion box show up in a corner. If 0.9.x still looked for motion and this isn't a change introduced in 0.10.0, maybe the dynamic contrast adjustment for motion detection is the culprit?

**Comment by blakeblackshear** [comment]
> Would it be possible to make motion a HA sensor entity that can be used to trigger HA event?

@tytlau You can use the `detection_fps` sensors in home assistant if you have detect on, but I am guessing you don't. Unfortuntely, there isn't a sensor for just motion, but you could open a new feature request issue.

**Comment by blakeblackshear** [comment]
> Is there a way now in 0.10.0 to disable everything for a camera, including motion?

There isn't. In 0.9.x, the detect switch also disabled motion detection. There isn't a new switch to disable motion detection, but you could open a feature request issue.

**Comment by tytlau** [comment]
> > Would it be possible to make motion a HA sensor entity that can be used to trigger HA event?
> 
> @tytlau You can use the `detection_fps` sensors in home assistant if you have detect on, but I am guessing you don't. Unfortuntely, there isn't a sensor for just motion, but you could open a new feature request issue.

@blakeblackshear thanks a lot! I have implemented as you have suggested and it's working well. I now have recording based on motion mode. When a motion is detected using detection fps counter, I will get a notification. When the camera detects a person (when lighting condition is better), it will still be marked in event.

**Comment by hawkeye217** [comment]
> Motion does not generate events in the events tab, but it will limit recordings to segments with motion. You can go back and review recordings and it will only show you sections of video with motion. The following config will keep 7 days of recordings, but only segments with actual motion:
> 
> ```yaml
> retain:
>   enabled: True
>   retain:
>     days: 7
>     mode: motion # <--- default
> ```
> 
> This works even when detect is disabled.

This is working great for me. 

Is it possible to set a ``pre_capture`` and ``post_capture`` time for motion events as well (not just object events)?

I can open a feature request if it isn't currently possible.

**Comment by blakeblackshear** [comment]
> Is it possible to set a `pre_capture` and `post_capture` time for motion events as well (not just object events)?

There isn't and I am not sure why you would want it. Said another way, you would be asking frigate to keep 10 extra seconds of video where nothing changed frame to frame. Frigate could also just show you 10s of the first frame and it should be the same thing. What am I missing?

**Comment by hawkeye217** [comment]
> > Is it possible to set a `pre_capture` and `post_capture` time for motion events as well (not just object events)?
> 
> There isn't and I am not sure why you would want it. Said another way, you would be asking frigate to keep 10 extra seconds of video where nothing changed frame to frame. Frigate could also just show you 10s of the first frame and it should be the same thing. What am I missing?

Fair point. In theory, you're totally right. 

Though when I look at my motion recordings with motion events back to back, I sometimes see a jump between when a car *should* be in the frame and when frigate detects it in the frame and begins the motion event. For example, the recording doesn't smoothly begin with a car appearing at the immediate edge of my camera's view, rather I see it when the car's front wheel is already well into the frame. 

I wonder if what I'm seeing could possibly be caused by a lower frame rate on the ``detect`` substream vs. the higher framerate on the ``record`` main stream.

I'm also being pretty picky... This new feature is awesome, and I'm happy to have it!

**Comment by blakeblackshear** [comment]
Could be a sync issue between the two streams. Frigate assumes that everything happens at the same time across those streams and it's possible that's not always the case. Note that the video segments are 10s long, so you are already getting some buffer on either side because frigate doesn't trim those segments at all. If motion happens in the last frame of the 10s, the entire segment is retained. In that case you get a 10s precapture.

**Comment by hawkeye217** [comment]
> Could be a sync issue between the two streams. Frigate assumes that everything happens at the same time across those streams and it's possible that's not always the case. Note that the video segments are 10s long, so you are already getting some buffer on either side because frigate doesn't trim those segments at all. If motion happens in the last frame of the 10s, the entire segment is retained. In that case you get a 10s precapture.

That's probably what it is, then.  And it's really no big deal. This is happening primarily on a really cheap cam and it would not surprise me if the thing doesn't keep synced between the streams. I had to tweak some of the ffmpeg parameters to get it to work at all.

Gives me another good reason to upgrade the camera to a nicer Dahua!

Thanks for the help.

**Comment by 18923** [comment]
Where in "frigate.yml" do I add the lines below so I can record on motion only on a particular camera. I tried in several camera subheadings but it always errors out and Frigate does not start.

retain:
  enabled: True
  retain:
    days: 7
    mode: motion # <--- default

**Comment by NickM-27** [comment]
> Where in "frigate.yml" do I add the lines below so I can record on motion only on a particular camera. I tried in several camera subheadings but it always errors out and Frigate does not start.
> 
> 
> 
> retain:
> 
>   enabled: True
> 
>   retain:
> 
>     days: 7
> 
>     mode: motion # <--- default


record:
  enabled: True
  retain:
    days: 7
    mode: motion

which can be set on a camera (so one indent inside of camera heading)

**Comment by 18923** [comment]
That makes sense, thank you.
The example had: (RETAIN) not record 
camera_name:
  record:   
    enabled: True
    retain:
      days: 7
      mode: motion

**Comment by peledzius** [comment]
> > Is it possible to set a `pre_capture` and `post_capture` time for motion events as well (not just object events)?
> 
> There isn't and I am not sure why you would want it. Said another way, you would be asking frigate to keep 10 extra seconds of video where nothing changed frame to frame. Frigate could also just show you 10s of the first frame and it should be the same thing. What am I missing?

Actually, I'm interested in this too. 

I'm currently using Frigate without detection. Recordings with audio makes sense to keep that extra pre/post motion. While watching recordings, in some situations I want to hear audio before motion happened. AFAIK, many DVRs/NVRs support pre/post motion capture.

**Comment by mannish-kumar** [comment]
I believe these motion recordings (without any object detected) are stored under the recordings folder. Correct me if I am wrong.

Where can I find the clips with motion where an object (like parson, cat etc.) was detected. Is it under the same recordings folder?

**Comment by NickM-27** [comment]
> I believe these motion recordings (without any object detected) are stored under the recordings folder. Correct me if I am wrong.
> 
> Where can I find the clips with motion where an object (like parson, cat etc.) was detected. Is it under the same recordings folder?

They're all in the same place in recordings directory

**Comment by mannish-kumar** [comment]
> 

From the YouTube videos, I see that when Frigate is integrated with HA these are presented as Clips, Recordings and Snapshots.
Is there a plan to present them in Frigate UI the same way as in HA? Separate links similar to events can be added for these.

The reason why I am asking is because it is very difficult to differentiate between these videos unless I open them individually and check. I don't have any plans to integrate with HA unless I am planning to run other Home Automation stuff.

I know frigate is meant to work best with HA integration. But It would be nice to have it in Frigate UI so that people like me who doesn't want HA can use Frigate independently.

**Comment by NickM-27** [comment]
> > 
> 
> From the YouTube videos, I see that when Frigate is integrated with HA these are presented as Clips, Recordings and Snapshots. Is there a plan to present them in Frigate UI the same way as in HA? Separate links similar to events can be added for these.
> 
> The reason why I am asking is because it is very difficult to differentiate between these videos unless I open them individually and check. I don't have any plans to integrate with HA unless I am planning to run other Home Automation stuff.
> 
> I know frigate is meant to work best with HA integration. But It would be nice to have it in Frigate UI so that people like me who doesn't want HA can use Frigate independently.

It's unclear what you are looking to do. The frigate `recordings/` directory is not meant to be interacted with manually. There are plenty of APIs available to get the clips you want and there is also the UI to download whatever clips from events you are looking for. Basically everything you've mentioned already exists so I am unclear what it is you want to do but feel you can't

**Comment by mannish-kumar** [comment]
>

**Comment by mannish-kumar** [comment]
Hi Nick,

As you can see in the below HA screenshot, HA has categorized the camera recordings into Clips(directory where an object is identified), Recordings (directory for videos where no active object was detected but motion was detected) and Snapshots (where snaps are stored).

Now if I have set "motion mode = all or motion" for the camera then it becomes easier for me to view through HA and I know that I will find the 24/7 recording/ motion events (no object detected) under Recordings directory.

![image](https://user-images.githubusercontent.com/87553561/192265036-01ca14c0-861f-4e1e-bb95-974fd980062c.png)

Now lets say I don't have Frigate integrated with HA then through Frigate GUI I am able to only see the events  where the objects were detected. For the other motion and 24/7 recordings I need to check the recordings directory and the directory has all the videos including events, 24/7 or motion as you can see in the screenshot. It becomes very difficult if I have to only see 24/7 or motion events(with no object identified)

I will be good if I can see the 24/7 or motion events in Frigate UI.

Hope I am clear this time.

![image](https://user-images.githubusercontent.com/87553561/192268909-58023004-c2fc-427c-b2fa-bc5805818707.png)

**Comment by mannish-kumar** [comment]
I checked that before writing here but for some reason it shows me only events where objects are identified. Check the events tab in right.
![image](https://user-images.githubusercontent.com/87553561/192281117-2fb2ee98-ee5c-4ea3-9096-b03e089790df.png)

**Comment by NickM-27** [comment]
Those are just event shortcuts (notice if you click on one it still shows a player for the entire hour and not just the single event)

You can still click on the hour above to start at the beginning of the hour and you can collapse the day and move between the days as well.

**Comment by doug62** [comment]
+1 - I Agree - would be nice to create an event that is "Motion" and have that follow the normal UI/Flow/Function of other events.

**Comment by mannish-kumar** [comment]
Thank You Nick, got it now. I wouldn't have known this without you helping me.
But I believe this should have been simpler or documented somewhere.

**Comment by distante** [comment]
Hi @blakeblackshear, Is the option to trigger motion events (without detection) in some of the betas? I am trying to get really sensitive motion-detection notifications on a specific part of a RTSP camera feed but I am not able to get any notification (in HA) unless I activate "Detect". (Using the stable release)

**Comment by NickM-27** [comment]
> Hi @blakeblackshear, Is the option to trigger motion events (without detection) in some of the betas? I am trying to get really sensitive motion-detection notifications on a specific part of a RTSP camera feed but I am not able to get any notification (in HA) unless I activate "Detect". (Using the stable release)

There is a motion sensor topic which shows when frigate detects motion. There are no motion events in an event timeline.

**Comment by distante** [comment]
> > Hi @blakeblackshear, Is the option to trigger motion events (without detection) in some of the betas? I am trying to get really sensitive motion-detection notifications on a specific part of a RTSP camera feed but I am not able to get any notification (in HA) unless I activate "Detect". (Using the stable release)
> 
> There is a motion sensor topic which shows when frigate detects motion, there is also a sensor in HA for this. There are no motion events in an event timeline.

Oh, I totally miss it, thank you @NickM-27! Will now try to find the best way to configure a really sensitive "move detected/stop movement" configuration.

**Comment by lolorc** [comment]
I guess an easy hack would be to use the motion history from the mqtt topic and produce a  html page with links to the correct video segments.
A bit weird this is still not a frigate feature

**Comment by incith** [comment]
> I guess an easy hack would be to use the motion history from the mqtt topic and produce a html page with links to the correct video segments.
> 
> A bit weird this is still not a frigate feature I've just had my first look at the code, what about adding a new object type called "motion" and register new events when motion is detected ?

This is why I'm here.  I'd love to see this, just a simple Motion event.  I thought that's how it would work if I set everything to just 'motion', and it is recording everything but it doesn't seem intuitive.  I was also originally doing recording for person and cat, and now since removing cat, everything is just detected as person.  And since person is the default, it becomes odd that 'motion' isn't recording all motion events when you explicitly leave out all references to detect people/etc.

Not sure if what I'm saying is coming across the way mean it to.  I'm sure we all have different ideas of what intuitive behavior would be for an app.

Anyway, tldr, just throwing my hat in the ring to add a singular motion event that will just record motion segments.  Of course it'll be up to the user to black out areas etc, and I would expect to have a fair number of motion events.

This project is truly amazing.  I absolutely love it.  Keep up the fantastic work.

**Comment by NickM-27** [comment]
> And since person is the default, it becomes odd that 'motion' isn't recording all motion events when you explicitly leave out all references to detect people/etc.

Events are tracked objects, you can still easily have motion based 24/7 recording along with events. 

A future version will allow seeing a timeline in the recordings view with the points in time where motion was detected shown and easily seekable.

**Comment by incith** [comment]
Right!  I guess that's what I was trying to agree with above, a motion event that gets logged to the same events page would be used by more than a few people I think.

I guess e.g my concern is if something amazing happens outside I don't want to miss it because a tracked object wasn't in the shot.

**Comment by NickM-27** [comment]
Yeah, the events page is for tracked objects. Motion detection is unreliable and would just clog the events with clutter and make it difficult to see the actually important things that happened.

**Comment by incith** [comment]
You know, I don't disagree.

I think my best use scenario is recording everything then with tracked objects.

Appreciate everything!  Thank you!

**Comment by NickM-27** [comment]
I'm a little confused by that last part so I want to make sure it's clear. You can easily record events with tracked objects AND record 24/7 on any motion (or record 24/7 and keep everything even if there was no motion). That way if something wasn't a tracked object but happened it was still recorded.

**Comment by incith** [comment]
Yup, exactly.  That's how I plan to do it now.  Thanks again!

**Comment by BBaoVanC** [comment]
> +1 this feature, image detection isn't good enough to track all that is needed - Record on any motion (within zones) would be a very critical feature.

Record on motion is already a thing, as in you can configure Frigate to not delete old footage past the 24/7 retention date if motion was detected.

For me, most of the motivation for wanting motion to show up in Events is gone if given the ability to create custom events  from the recordings view (which iirc is coming in the next release) for stuff that didn't get detected, and bonus if there were a nicer UI for recordings.

**Comment by doug62** [comment]
+1 - I created a similar request quite some time ago, I have so many false positives + true negatives with image detection that I really can only count on motion, I actually had a car stolen but that wasn't detected...
If someone could create a sensor/class called "motion" and have that work within the same object detection flow that would be brilliant.  
I wonder if my problems are that I use CPU rather than a card for detection, I run this in a K8 cluster with many CPUs and if quality is based on available CPU, that ain't my problem? (i.e. does this use CPU threads properly?)

**Comment by NickM-27** [comment]
> +1 - I created a similar request quite some time ago, I have so many false positives + true negatives with image detection that I really can only count on motion, I actually had a car stolen but that wasn't detected...
> 
> If someone could create a sensor/class called "motion" and have that work within the same object detection flow that would be brilliant.  
> 
> I wonder if my problems are that I use CPU rather than a card for detection, I run this in a K8 cluster with many CPUs and if quality is based on available CPU, that ain't my problem? (i.e. does this use CPU threads properly?)

It doesn't matter how fast the CPU is, it is still orders of magnitude slower than a coral or GPU. This is likely to be an issue on your setup.

**Comment by doug62** [comment]
@NickM-27 Re: "It doesn't matter how fast the CPU"  I have 6 cams, never seems to use more than 6CPUs when 32 are available.... So, if my poor detection is caused by resource it might be caused by software not threading/scaling optimally.

**Comment by NickM-27** [comment]
> @NickM-27 Re: "It doesn't matter how fast the CPU"  I have 6 cams, never seems to use more than 6CPUs when 32 are available.... So, if my poor detection is caused by resource it might be caused by software not threading/scaling optimally.

That's not how it works, doesn't matter if your CPU is not being fully utilized, CPUs run the inferences slowly because it requires more instructions. Meanwhile a tpu or GPU runs these operations. 

Same thing about why a gpu decodes / encodes video faster than a CPU.

**Comment by doug62** [comment]
@NickM-27 I'm wondering why object detection is so unreliable, which is why I'm hoping we can make motion detection a first class citizen.  I was GUESSING that poor object detection might be caused by not utilizing available compute, perhaps because the code doesn't utilize threads/maximum CPU/resources?  (From what I have seen, FFMEP/OpenCV don't do this well)

Take this line for example: (I'm a C# guy, this is PLINQ)
foreach (var frame in frames.AsParallel().WithDegreeOfParallelism(50))
{
   ProcessFrame(frame);
}

This code instantly starts 50 threads on 50 cores if they are there (otherwise it will mange them).  I have 200 Cores in a Kubernetes cluster, why would I pay for custom hardware?

I see that you are a Sponsor/Collaborator - Thanks for you good work here

**Comment by NickM-27** [comment]
I don't see how I can explain it more clearly, extra cores doesn't help, dedicated hardware made for the task does. You can easily see this with the inference times on the system page and in the docs https://docs.frigate.video/frigate/hardware#detectors

**Comment by NickM-27** [comment]
Another part of explanation for suboptimal object detection is that the default model is a Google demo model trained on the COCO dataset. [If you look at some examples of images in the COCO dataset](https://imgur.com/a/CNImofJ) they do not look like the security camera images at all. This leads to false positives. 

There are a number of filters and other options to reduce false positives. 

Better models, like [what Frigate+ is planning to offer](https://docs.frigate.video/integrations/plus), that are trained on actual security camera examples offer higher accuracy and less issues with false positives.

**Comment by doug62** [comment]
I think we are saying the same thing - extra cores don't help cos the software doesn't take advantage of available resources.  AND proprietary hardware is required for object detection.

**Comment by BBaoVanC** [comment]
> I think we are saying the same thing - extra cores don't help cos the software doesn't take advantage of available resources. AND proprietary hardware is required for object detection.

If I understand correctly, "slowly" is referring to the latency. With software video encoding for example, adding more CPU cores doesn't really make it process any individual part of the video faster, instead it just processes more of the video simultaneously. So I think the object detection is similar. With more cores, it will still take the same 100ms to run the detection, but you'd be able to do more cameras simultaneously.

So it can't take advantage of more available resources because there's no more work left to feed to it that isn't already in process.

**Comment by NickM-27** [comment]
> > I think we are saying the same thing - extra cores don't help cos the software doesn't take advantage of available resources. AND proprietary hardware is required for object detection.
> 
> If I understand correctly, "slowly" is referring to the latency. With software video encoding for example, adding more CPU cores doesn't really make it process any individual part of the video faster, instead it just processes more of the video simultaneously. So I think the object detection is similar. With more cores, it will still take the same 100ms to run the detection, but you'd be able to do more cameras simultaneously.
> 
> So it can't take advantage of more available resources because there's no more work left to feed to it that isn't already in process.

exactly, running a single model inference is a synchronous action, multi threading does not improve this. It simply requires many more instructions on a general computing device (CPU) than it does on a device dedicated to these inferences.

**Comment by doug62** [comment]
@NickM-27 Sorry to keep nagging, Noticed that my Frigate reports 103% utilization while only using one CPU while about 20 are available, I really want this to work, since you have more experience with the code, could you point me to the file that does the frame processing?  I would like to take a crack at fixing the multi-threading area so that this could work with more than a 3 cameras on a high power PC.....  Thanks in advance.

**Comment by ivanjx** [comment]
this would be far more useful for my use case since the detection AI model is not good enough with my camera. better to have false positives than false negatives.

**Comment by vw-kombi** [comment]
For my config, I have a front camera with three zones.  Front Left, Front Path and driveway.
I want people detection and recording events on Front Left and Driveway, but the front path which includes my letterbox never records the postman as they are on a strange electric bike contraction and this is never identified as anything, so I dont get any recordings of deliveries.  So I want this zone to record on ANY MOVEMENT.  Its important to get deliveries so when they say something was delivered, I can prove it was not.  Any ideas ?

**Comment by NickM-27** [comment]
https://docs.frigate.video/configuration/record#reduced-storage-only-saving-video-when-motion-is-detected

**Comment by vw-kombi** [comment]
> https://docs.frigate.video/configuration/record#reduced-storage-only-saving-video-when-motion-is-detected

Hi - Is this directed at my comment as I see nothing in there that will help what I want to do - or I am not understanding it.......

**Comment by NickM-27** [comment]
You said

> So I want this zone to record on ANY MOVEMENT.

the docs I linked to explain directly how to configure frigate to record on any movement

**Comment by vw-kombi** [comment]
I am sorry - I had a stroke some time back and it was a while of googling to get what I have now working almost perfectly (just missing the postman on his cart).  Maybe all the config stuff I picked up along the way is conflicting here.  This is my config below, and its the front-path zone that I want to not just be person, but any movement, then create a snapshot, event and start recording while there is movement :

cameras:
  Front-Cam:
    enabled: true
    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
      - path: rtsp://hikvision:password@10.10.10.18:554/Streaming/Channels/102   # <----- The stream you want to use for detection
        input_args: preset-rtsp-restream
        roles:
        - detect
        - rtmp
      - path: rtsp://hikvision:password@10.10.10.18:554/Streaming/Channels/101   # <----- The stream you want to use for detection
        input_args: preset-rtsp-restream
        roles:
        - record
    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 1280
      height: 720
      fps: 6
    objects:
      track:
      - person
      - motorcycle
      filters:
        motorcycle:
          min_score: 40
    record:
      enabled: true
      retain:
        days: 7
        mode: motion
      events:
        required_zones:
        - front-left
        - front-path
        - driveway
        retain:
          default: 7
          mode: motion
    snapshots:
      enabled: true
      required_zones:
      - front-left
      - front-path
      - driveway
      bounding_box: true
      timestamp: true
      retain:
        default: 7
    motion:
      mask:
      - 184,676,426,673,425,720,0,720,0,671
      - 1024,0,1024,31,711,28,711,0
      - 967,720,1024,720,1024,34,649,27,643,47,879,92,842,282
      - 327,375,683,462,670,222,548,190,418,182
      - 0,372,155,410,182,315,221,179,112,177
      - 0,204,128,108,129,0,0,0
    zones:
      driveway:
        coordinates: 0,365,0,664,407,663,418,692,673,694,690,210,247,171
      front-path:
        coordinates: 
          863,117,855,217,828,280,851,346,929,634,873,720,693,720,708,414,708,135,530,96,533,57
      front-left:
        coordinates: 196,105,338,89,343,49,192,68

**Comment by NickM-27** [comment]
>  then create a snapshot, event and start recording while there is movement :

right, I think this is the disconnect here. That is not supported like it says above. In frigate 0.13 you will only get recordings for this scenario which you will need to view using the recordings viewer 

![Screen Shot 2024-03-20 at 16 08 47 PM](https://github.com/blakeblackshear/frigate/assets/14866235/71615701-166a-4db7-9807-cbdfac4518a2)

This issue is closed because in the next version of frigate (0.14) the UI has been rewritten and there will be a dedicated place to view a timeline showing motion activity and times where objects were detected so it will be easy to review things that frigate did not detect.
